### PR TITLE
Configuring field size based on an argument for the ErForceSimulator constructor

### LINKED ISSUE #2341: ErForceSimulator: Configure the field size based on the constructor argument
### Description of the task
Currently, the ErForceSimulator defaults to a DivA field. Instead it should depend on a constructor argument passed into the simulator, perhaps `field`.

### Acceptance criteria

- [ ] Configure the field size for the ErForceSimulator based on a constructor argument

### PR DESCRIPTION
**Pull request by HubertTheodore**
### Description

Made changes to the ErForceSimulator constructor so that it can take a field type argument.

### Resolved Issues

resolves #2341 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/simulation/er_force_simulator.cpp [hunk 1]
```diff
@@ -17,16 +18,27 @@
...
-    QString full_filename = CONFIG_DIRECTORY + CONFIG_FILE + ".txt";
+    QString full_filename = CONFIG_DIRECTORY;
+
+    if (field_type == FieldType::DIV_A)
+    {
+        full_filename = full_filename + CONFIG_FILE + ".txt";
+    }
+    else
+    {
+        full_filename = full_filename + CONFIG_FILE + "B.txt";
```
Comment (jonathanlew) [inline src/software/simulation/er_force_simulator.cpp:39]: please add a comment to say we're loading the division B configuration here

File: src/software/simulation/er_force_simulator.cpp [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #include <QtCore/QString>
 #include <iostream>
 
+#include "extlibs/er_force_sim/src/amun/simulator/simulator.h"
```
Comment (jonathanlew) [inline src/software/simulation/er_force_simulator.cpp]: I don't think this is needed

File: src/software/world/field.h [hunk 3]
```diff
@@ -47,6 +50,13 @@ class Field
      */
     static Field createSSLDivisionAField();
 
+    /**
+     * Creates a field with the standard SSL Division A or B dimensions
+     *
+     * @return a field with the standard SSL Division A or B dimensions
+     */
+    static Field createField(FieldType field_type);
```
Comment (jonathanlew) [inline src/software/world/field.h:58]: please write a test for this function

File: src/software/simulation/er_force_simulator_test.cpp [hunk 4]
```diff
@@ -78,10 +77,14 @@ TEST_F(ErForceSimulatorTest, set_ball_state_when_ball_already_exists)
...
                     double y_difference = fabs(ball.y() - 20.0f);
-                    if (x_difference < 1.0f && y_difference < 1.0f)
+                    if (x_difference < 2.0f && y_difference < 2.0f)
                     {
                         at_least_one_wrapper_packet_passes = true;
                     }
+                    else
+                    {
+                        std::cout << x_difference << "," << y_difference << std::endl;
+                    }
```
Comment (jonathanlew) [inline src/software/simulation/er_force_simulator_test.cpp]: please remove this debug code

File: src/software/simulation/er_force_simulator.cpp [hunk 5]
```diff
@@ -226,7 +238,7 @@ std::vector<SSLProto::SSL_WrapperPacket> ErForceSimulator::getSSLWrapperPackets(
 
 Field ErForceSimulator::getField() const
 {
-    return Field::createSSLDivisionAField();
+    return field;
```
Comment (jonathanlew) [inline src/software/simulation/er_force_simulator.cpp:241]: please add a test for this function, i.e. create an ErForceSimulator with DIV_A/DIV_B and check the Field returned by getField.

**Review by jonathanlew — CHANGES_REQUESTED** [review]
overall looks good, just missing a bit of testing

File: src/software/simulation/er_force_simulator.h [hunk 6]
```diff
@@ -37,7 +37,8 @@ class ErForceSimulator : public QObject
      * @param wheel_constants The wheel constants
      * @param simulator_config The config to fetch parameters from
      */
-    explicit ErForceSimulator(const Field& field, const RobotConstants_t& robot_constants,
+    explicit ErForceSimulator(const FieldType& field_type,
```
Comment (jonathanlew) [inline src/software/simulation/er_force_simulator.h:41]: please update the javadoc above

File: src/software/world/field.h [hunk 7]
```diff
@@ -47,6 +50,13 @@ class Field
      */
     static Field createSSLDivisionAField();
 
+    /**
+     * Creates a field with the standard SSL Division A or B dimensions
+     *
+     * @return a field with the standard SSL Division A or B dimensions
```
Comment (jonathanlew) [inline src/software/world/field.h]: please add an `@param` for this javadoc

**Review by jonathanlew — CHANGES_REQUESTED** [review]
looks good, just a couple nits

**Review by jonathanlew — APPROVED** [review]
LGTM! :field_hockey:

**Review by jameschuang2002 — APPROVED** [review]

**Pull request merged**